### ISSUE TITLE
Focus.Quests ignores dungeon fight skip settings when shadow quest is present for regular dungeons

### ISSUE THREAD

**Issue by tauqua** [Farigh/pokeclicker-automation#359]
### Describe the bug
When using Focus.Quests, if the quests "Defeat the \<Dungeon\> \<n\> times" and "Catch \<n\> Shadow Pokémon" are both active, the dungeon will be fully cleared regardless of what the setting is for "Skip as many fights as possible" even when there are no Shadow Pokémon in the given dungeon.

### Steps to Reproduce
1. In the dungeon fight settings, enable "Skip as many fights as possible"
2. Accept a "Defeat the \<Dungeon\> \<n\> times" quest
3. Accept a "Catch \<n\> Shadow Pokémon" quest
4. Enable quests focus

### Expected behavior
* In dungeons with Shadow Pokémon, I would expect the "Skip as many fights as possible" option to have a "Disabled for the following reasons" tooltip to communicate why fights are not being skipped
* In dungeons without Shadow Pokémon, I would expect the "Skip as many fights as possible" option to be respected and not overridden.

**Comment by Farigh** [comment]
Hi, 

Sorry for the late reply, I was on vacation.

This can happen if you have a `Catch <n> Shadow Pokémon.` active at the same time.
It will try to catch those at the same time it completes the dungeon.

I could only enable this behaviour if the dungeon actually contains shadow pokemons to avoid fighting any pokémon otherwise.

Is it your case ? If not, I'll need a save reproducing the problem and a screenshot of your settings (no better ways until I implement #246 ...)

**Comment by tauqua** [comment]
Yes, this is the situation. I didn't realize initially, but updated the issue when I did; it should reflect the shadow issue as you're describing.

I think only enabling if the dungeon contains shadow is the best fix here.

**Comment by Farigh** [comment]
It Should now work has intended 😉
